Makefile

```diff
@@ -77,92 +84,108 @@ check-license: addlicense ## Check that every file has a license header present.
-	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" go run github.com/onsi/ginkgo/v2/ginkgo --label-filter="!integration" -coverprofile cover.out ./...
+test: manifests generate fmt vet envtest ## Run tests. Some test depend on Linux OS
+	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) --bin-dir $(LOCALBIN) -p path)" go run github.com/onsi/ginkgo/v2/ginkgo run -r --label-filter="!integration" -coverprofile cover.out
 
 .PHONY: integration-tests
 integration-tests: ## Run integration tests against code. For dependencies, refer to the integration-test workflow.
-	go run github.com/onsi/ginkgo/v2/ginkgo --label-filter="integration" -coverprofile cover.out ./...
+	go run github.com/onsi/ginkgo/v2/ginkgo run -r --label-filter="integration" -coverprofile cover.out
 
 ##@ Documentation
 
 .PHONY: start-docs
 start-docs: ## Start the local mkdocs based development environment.
-	docker build -t ${MKDOCS_IMG} -f docs/Dockerfile .
-	docker run -p 8000:8000 -v `pwd`/:/docs ${MKDOCS_IMG}
+	$(CONTAINER_TOOL) build -t ${MKDOCS_IMG} -f docs/Dockerfile .
+	$(CONTAINER_TOOL) run -p 8000:8000 -v `pwd`/:/docs ${MKDOCS_IMG}
 
 .PHONY: clean-docs
 clean-docs: ## Remove all local mkdocs Docker images (cleanup).
-	docker container prune --force --filter "label=project=libvirt-provider_documentation"
+	$(CONTAINER_TOOL) container prune --force --filter "label=project=libvirt-provider_documentation"
 
 ##@ Build
 
 .PHONY: build
-build: generate fmt add-license lint ## Build the binary
-	GOOS=$(TARGET_OS) GOARCH=$(TARGET_ARCH) go build -o LIBVIRTPROVIDERBIN $(LIBVIRTPROVIDERMAINPATH)/main.go
+build: manifests generate fmt vet add-license lint ## Build the binary
+	GOOS=$(TARGET_OS) GOARCH=$(TARGET_ARCH) CGO_ENABLED=$(CGO_ENABLED) go build -o $(LIBVIRT_PROVIDER_BIN) $(LIBVIRT_PROVIDER_BIN_SOURCE)
 
 .PHONY: run
-run: generate fmt lint ## Run the binary
-	go run $(LIBVIRTPROVIDERMAINPATH)/main.go
+run: manifests generate fmt vet ## Run the binary
+	go run $(LIBVIRT_PROVIDER_BIN_SOURCE)/main.go \
+		--supported-machine-classes config/development/machineclasses.json \
+		--network-interface-plugin-name isolated \
+		--address iri.sock
 
 .PHONY: docker-build
-docker-build: test ## Build docker image with partitionlet
-	docker build $(DOCKER_BUILDARGS) -t ${IMG} $(GITHUB_PAT_MOUNT) .
+docker-build: ## Build docker image with partitionlet
+	$(CONTAINER_TOOL) build $(CONTAINER_BUILDARGS) -t ${IMG} $(GITHUB_PAT_MOUNT) .
 
 .PHONY: docker-push
 docker-push: ## Push docker image with the manager.
-	docker push ${IMG}
+	$(CONTAINER_TOOL) push ${IMG}
 
 ##@ Deployment
+
+ifndef ignore-not-found
+  ignore-not-found = false
+endif
+
 .PHONY: deploy
 deploy: kustomize ## Deploy libvirt-provider into the K8s cluster specified in ~/.kube/config.
 	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
-	kubectl apply -k config/default
+	$(KUSTOMIZE) build config/default | $(KUBECTL) apply -f -
```

Comment:
Is there a reason for chaining the commands?

Reply:
i followed kubebuilder approach here: https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugins/golang/v4/scaffolds/internal/templates/makefile.go#L223